rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -69,21 +69,44 @@
 
 namespace
 {
+//==============================================================================
+// Helper function to create a copy of a string after removing any
+// leading or trailing slashes.
+std::string strip_slashes(const char * const str)
+{
+  std::string ret = std::string(str);
+  const std::size_t len = strlen(str);
+  std::size_t start = 0;
+  std::size_t end = len - 1;
+  if (str[0] == '/') {
+    ++start;
+  }
+  if (str[end] == '/') {
+    --end;
+  }
+  return ret.substr(start, end - start);
```

Comment:
I think this is incorrect.  Consider a str being passed in here that is "/foo".  In that case, `len` will be 4, `start` will begin at 0, and end will begin at 3.  Now `str[0]` is indeed `/`, so start is now `1`.  When we do the `ret.substr`, we'll only copy in `end - start`, `3 - 1`, or 2 bytes.  So that means we'll only get `fo`, when we meant to get `foo`.  Thus I think this needs to be:

```suggestion
  return ret.substr(start, end - start + 1);
```